router/router.go

```diff
@@ -1627,7 +1641,15 @@ func (rt *HandleT) commitStatusList(responseList *[]jobResponseT) {
 		}
 		//Update the status
 		err := rt.jobsDB.WithUpdateSafeTx(func(tx jobsdb.UpdateSafeTx) error {
-			err := rt.jobsDB.UpdateJobStatusInTx(tx, statusList, []string{rt.destName}, nil)
+			// The status of drained jobs (inside the rt_job_status* table) is
+			// updated inside the readAndProcess function
+			var nonDrainedStatusList []*jobsdb.JobStatusT
```

Comment:
Is this kind of filtering necessary? Under which conditions do we end up calling `commitStatusList` for drained job statuses?
As far as I can tell, we do not push drained job statuses in the `responseQ` channel.

_**Update:**_ just went a bit further and understood that we are explicitly sending now drained job statuses to the `responseQ` channel, not sure this is the best design choice though.

Reply:
Update: In the latest commits, I have changed the implementation to directly update the report inside the readAndProcess function